custom_components/waste_collection_schedule/waste_collection_schedule/source/publidata_ca.py

```diff
@@ -0,0 +1,482 @@
+import calendar
+import re
+from datetime import datetime, timedelta, timezone
+
+import requests
+from dateutil.rrule import (
+    DAILY,
+    FR,
+    MO,
+    MONTHLY,
+    SA,
+    SU,
+    TH,
+    TU,
+    WE,
+    WEEKLY,
+    rrule,
+    rruleset,
+)
+from waste_collection_schedule import Collection
+from waste_collection_schedule.exceptions import SourceArgumentException
+
+TITLE = "Publidata (Canada) generic source"
+DESCRIPTION = "Publidata is a Canadian public operator. Check if your area is concerned on their website."
+URL = "https://www.publidata.ca"
+COUNTRY = "ca"
+
+TEST_CASES = {
+    "Sainte-Julie City Hall": {
+        "address": "1580 Chem. du Fer-à-Cheval, Sainte-Julie, QC J3E 0A2",
+    },
+    "Calixa-Lavallée City Hall": {
+        "address": "771 Chem. de la Beauce, Calixa-Lavallée, QC J0L 1A0"
+    },
+    "Contrecoeur City Hall": {
+        "address": "5000 Rte Marie-Victorin, Contrecoeur, QC J0L 1C0"
+    },
+    "Saint-Amable City Hall": {
+        "address": "575 Rue Principale, Saint-Amable, QC J0L 1N0"
+    },
+    "Varennes City Hall": {
+        "address": "175 Rue Sainte-Anne, Varennes, QC J3X 1R6"
+    },
+    "Verchères City Hall" : {
+        "address": "581 QC-132, Verchères, QC J0L 2R0"
+    }
```

Comment:
Maybe reduce your test cases to 3 and comment out the rest. Having a lot of tests, cases accumulates when testing all sources

Reply:
Done. Kept the first three.